miasm2/os_dep/common.py

```diff
@@ -73,10 +74,13 @@ def vm_alloc(self, vm, size, perm=PAGE_READ | PAGE_WRITE):
             combination of them); default is PAGE_READ|PAGE_WRITE
         """
         addr = self.next_addr(size)
-        vm.add_memory_page(addr, perm, "\x00" * size,
+        vm.add_memory_page(addr, perm, "\x00" * (size),
                            "Heap alloc by %s" % get_caller_name(2))
         return addr
 
+    def get_size(self, vm, ptr):
+        return vm.get_all_memory()[ptr]["size"]
```

Comment:
As you add a new public function, can you add a docstring describing it?
I know this is not the case for every function in this module, but we (at least try to) add docstring on new implemented function. Thus, Miasm readibility gets better on average :smiley: 

For this function, there is a particularity: `ptr` must be the base address of the mapped function. If you want to have a more permissive function, you can do something like (untested but should work):
```Python
assert vm.is_mapped(ptr, 1)
data = vm.get_all_memory()
ptr_page = data.get(ptr, None)
if ptr_page is None:
    for address, page_info in data.iteritems():
        if address <= ptr < address + page_info["size"]:
            ptr_page = page_info
            break
    else:
        raise RuntimeError("Must never happen (unmapped but mark as mapped by API)")
return ptr_page["size"]
```
Note for a future PR: this function should probably reside in VmMngr implementation

Reply:
I think the semantic of the function should be indeed more clear. About what you propose, I think we should actually make 2 functions : ``get_page_size``, which asserts that the ``ptr`` is the start address of a page, and a ``get_alloc_size``, which does what you do and returns the number of available bytes starting from ``ptr`` (so that's actually a bit different from your implementation, but as long as it's properly documented).  What do you think?

Reply:
It could be confusing (between `get_page_size` / `get_alloc_size`). In addition, we do not have any use of a `get_alloc_size` as you describe for now.
So I'll prefer a unique `get_page_size` API. If you find it confusing to be able to get the allocated page size for a address in the middle of the page, drop my comment but make it clear in the docstring (something like "@ptr: starting address of the target page")

Reply:
Hmm indeed the usage of ``get_alloc_size`` is really limited. Your proposal is fine to me as long as it's also stated in the documentation. I'll do it this way thus :)